cloudify_deployment_proxy/polling.py

```diff
@@ -190,14 +191,29 @@ def dep_workflow_in_state_pollster(_client,
             'Executions list failed {0}.'.format(str(ex)))
     else:
         for _exec in _execs:
+            ctx.logger.debug(
+                '_exec info id {0}'.format(_exec.get('id')))
+
+            ctx.logger.debug(
+                '_exec info state {0}'.format(_exec.get('status')))
+
             if _workflow_id and _exec.get('workflow_id', '') != _workflow_id:
                 continue
             if _log_redirect:
                 ctx.logger.debug(
                     '_exec info for _log_redirect is {0}'.format(_exec))
                 dep_logs_redirect(_client, _exec.get('id'))
+
             if _exec.get('status') == _state:
-                return True
+                # After checking the status, another check needed which is
+                # to check if the ``_execution_id`` is set or not and
+                # whether or not the ``_execution_id`` is the same as the
+                # one returned from the api
+                if not _execution_id or _execution_id == _exec.get('id'):
```

Comment:
if not _execution_id --> raise NonRecoverableError / RuntimeError before loop start rather than skip. I think that execution ID is most important argument for poller because it uniquely identifies single execution.

Reply:
Done